BrushedMotorControllerFirmware/include/driver.c

```diff
@@ -220,25 +274,52 @@ static void continuous_calculation_timer_handler(struct k_timer *dummy)
 K_TIMER_DEFINE(continuous_calculation_timer, continuous_calculation_timer_handler, NULL);
 #pragma endregion TimerWorkCallback
 
-// encoder functions
-static void enc_callback(enum ChannelNumber chnl, enum PinNumber pin)
+bool is_target_achieved(enum ChannelNumber chnl)
 {
-	// TODO - maybe split this callback to two callbacks, for P1 and P0:
-	// what if motor is stuck and hitting one enc. from two directions?
-	// How will speed calculation behave if motor will keep changing position?
-	// (ex. when setting position?)
-	drv_chnls[chnl].count_cycles += 1;
+	uint32_t delta_shortest_path;
+
+	CALC_POSITION_DELTA(chnl)
 
-	if (drv_chnls[chnl].actual_mrpm == 0) {
-		drv_chnls[chnl].actual_dir = STOPPED;
-		return;
+	if (drv_chnls[chnl].position_delta < HALF_SPIN_DEGREES) {
+		delta_shortest_path = drv_chnls[chnl].position_delta;
+	} else {
+		delta_shortest_path = ((int32_t)FULL_SPIN_DEGREES -
+					drv_chnls[chnl].position_delta);
 	}
 
+	if (drv_chnls[chnl].target_achieved) {
+		if (delta_shortest_path <=
+			(FULL_SPIN_DEGREES) / (CONFIG_POS_CONTROL_PRECISION_MODIFIER)) {
+			return true;
+		}
+	} else {
+		if (delta_shortest_path <=
+		   ((FULL_SPIN_DEGREES * CONFIG_POS_CONTROL_HISTERESIS_PERCENTAGE) /
+		    (CONFIG_POS_CONTROL_PRECISION_MODIFIER * 100))) {
+
+			return true;
+		}
+	}
+
+
+
+	return false;
+}
+
+// encoder functions
+static void enc_callback(enum ChannelNumber chnl, enum PinNumber pin)
+{
 	if (pin == P0) {
 		if (drv_chnls[chnl].prev_val_enc[P0] == drv_chnls[chnl].prev_val_enc[P1]) {
-			drv_chnls[chnl].actual_dir = BACKWARD;
+			drv_chnls[chnl].count_cycles_bck += 1;
 		} else {
-			drv_chnls[chnl].actual_dir = FORWARD;
+			drv_chnls[chnl].count_cycles_fwd += 1;
+		}
+	} else if (pin == P1) {
+		if (drv_chnls[chnl].prev_val_enc[P0] == drv_chnls[chnl].prev_val_enc[P1]) {
+			drv_chnls[chnl].count_cycles_fwd += 1;
+		} else {
+			drv_chnls[chnl].count_cycles_bck += 1;
```

Comment:
```suggestion
			drv_chnls[chnl].count_cycles_fwd++;
		}
	} else if (pin == P1) {
		if (drv_chnls[chnl].prev_val_enc[P0] == drv_chnls[chnl].prev_val_enc[P1]) {
			drv_chnls[chnl].count_cycles_fwd++;
		} else {
			drv_chnls[chnl].count_cycles_bck++;
```

Reply:
Corrected in https://github.com/eSqadron/NRF52840-Brushed-Motor-Controller-Firmware/pull/13